yt_geotiff/utilities.py

```diff
@@ -127,4 +132,37 @@ def merge_dicts(*dict_args):
     for dictionary in dict_args:
         result.update(dictionary)
     return result
-
+      
+
+def rasterio_window_calc(selector):
+       """ 
+       This function reads information from either a sphere, box or region selector 
+       object and outputs the dimensions of a container: left edge, right edge,
+       width and height.
+       
+       Output uses:
+              - Construct dimensions to perform a Rasterio Window Read
+              - Provide a temporary update ro the image domain size             
+       """        
+       #pdb.set_trace()
+       if type(selector) == selector_shape.SphereSelector:
```

Comment:
This is a more robust way to check that we have the right type. As well, if I make a circle class that is a subclass of sphere, it will catch that as well.
```suggestion
       if isinstance(selector, selector_shape.SphereSelector):
```